shell/platform/darwin/ios/framework/Source/FlutterStandardCodec.mm

```diff
@@ -36,34 +55,53 @@ - (id)decode:(NSData*)message {
 
 #pragma mark - Codec for method channel
 
-@implementation FlutterStandardMethodCodec
+@implementation FlutterStandardMethodCodec {
+  FlutterStandardReaderWriter* _readerWriter;
+}
 + (instancetype)sharedInstance {
   static id _sharedInstance = nil;
   if (!_sharedInstance) {
-    _sharedInstance = [FlutterStandardMethodCodec new];
+    _sharedInstance =
+        [FlutterStandardMethodCodec withReaderWriter:[FlutterStandardReaderWriter new]];
```

Comment:
Lets make the `FlutterStandardMethodCodec` take ownership of `FlutterStandardReaderWriter` and then assign it to the shared instance. As written, `FlutterStandardReaderWriter` is leaked and `FlutterStandardMethodCodec` autoreleased after assigning to `_sharedInstance` which will make that reference garbage after the ARP activation.

Instead of using the `new`, let's be explicit and specify `alloc-init` instead (and in this case, the `autorelease` that you missed).